@here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts

```diff
@@ -910,41 +944,45 @@ export class OmvDecodedTileEmitter implements IOmvEmitter {
                             vertices[i + 1],
                             vertices[i + 2] + minHeight * scaleFactor
                         );
+                        if (texCoordType !== undefined) {
+                            textureCoordinates.push(vertices[i + 3], vertices[i + 4]);
+                        }
+
                         if (isExtruded) {
                             positions.push(
                                 vertices[i],
                                 vertices[i + 1],
                                 vertices[i + 2] + height * scaleFactor
                             );
-                        } else if (isTextured) {
-                            textureCoordinates.push(vertices[i + 3], vertices[i + 4]);
+                            if (texCoordType !== undefined) {
+                                textureCoordinates.push(vertices[i + 3], vertices[i + 4]);
+                            }
                         }
                     }
 
                     // Add the footprint/roof indices to the index buffer.
                     const vertexStride = isExtruded ? 2 : 1;
                     for (let i = 0; i < triangles.length; i += 3) {
-                        const v0 = baseVertex + triangles[i + 0] * vertexStride;
-                        const v1 = baseVertex + triangles[i + 1] * vertexStride;
-                        const v2 = baseVertex + triangles[i + 2] * vertexStride;
-                        indices.push(v0, v1, v2);
-
                         if (isExtruded) {
```

Comment:
Each loop iteration will only add 3 indices, however for isExtruded, it should add 6 no?

Reply:
No. We don't render the footprint, but only the roof.